redis/redis_test.go

```diff
@@ -0,0 +1,60 @@
+package redis
+
+import (
+	"log"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/alicebob/miniredis"
+	"github.com/elliotchance/redismock"
+	"github.com/go-redis/redis"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	client *redis.Client
+)
+
+var (
+	key = "key"
+	val = "val"
+)
+
+func TestMain(m *testing.M) {
+	// Start redis mock server
+	mr, err := miniredis.Run()
+	if err != nil {
+		log.Fatalf("an error '%s' was not expected when opening a stub database connection", err)
+	}
+
+	// New redis client point to mock server
+	client = redis.NewClient(&redis.Options{
+		Addr: mr.Addr(),
+	})
+
+	// Assert exit code
+	code := m.Run()
+	os.Exit(code)
```

Comment:
you can't exit in a test.
I don't get the purpose of this test, there is no assertion